nixos/modules/services/networking/bee-clef.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.bee-clef;
+in {
+  meta = {
+    maintainers = with maintainers; [ attila-lendvai ];
+  };
+
+  ### interface
+
+  options = {
+    services.bee-clef = {
+      enable = mkEnableOption "clef external signer instance for Ethereum Swarm Bee";
+
+      data-dir = mkOption {
+        type = types.nullOr types.str;
+        default = "/var/lib/bee-clef";
+        description = "Data dir for bee-clef.";
+      };
+
+      password-file = mkOption {
+        type = types.nullOr types.str;
+        default = "/var/lib/bee-clef/password";
+        description = "Password file for bee-clef.";
+      };
+    };
+  };
+
+  ### implementation
+
+  config = mkIf cfg.enable {
+    systemd.packages = [ pkgs.bee-clef ];
+    # if we ever want to have rules.js under /etc/bee-clef/
+    # environment.etc."bee-clef/rules.js".source = ${pkgs.bee-clef}/rules.js
+
+    services.bee-clef = {
+      data-dir      = "/var/lib/bee-clef";
+      password-file = "/var/lib/bee-clef/password";
+    };
+
+    systemd.tmpfiles.rules = [
+        "d '${cfg.data-dir}' 0750 bee-clef bee-clef"
+        "d '${cfg.data-dir}/keystore' 0700 bee-clef bee-clef"
+      ];
+
+    systemd.services.bee-clef = {
+      wantedBy = [ "bee.service" ];
+      serviceConfig = {
+        ExecStart = [
+          "${pkgs.bee-clef}/clef-service start"
+        ];
+        ExecStop = [
+          "${pkgs.bee-clef}/clef-service stop"
+        ];
```

Comment:
Can both be a single string.

Reply:
what do you mean by that? (i'm no expert of systemd)

Reply:
@attila-lendvai if there is only a single command to run in an `ExecStart` or `ExecStop` you can specify a string **or** a list of strings, like so:
```
ExecStart = "${pkgs.bee-clef}/clef-service start";
```

So what you had is fine, or what I have shown above is fine. Typically it makes more sense to use a single string approach if you aren't passing a list. But like I said... no longer relevant.